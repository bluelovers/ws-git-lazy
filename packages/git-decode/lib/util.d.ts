/**
 * Created by user on 2020/6/4.
 */
export declare function isQuote(input: string): boolean;
export declare function removeQuote(input: string): string;
export declare function wrapQuote(input: string, quote?: string): string;
