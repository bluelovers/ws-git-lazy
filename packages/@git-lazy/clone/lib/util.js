"use strict";
/**
 * Created by user on 2020/6/15.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.gitCloneCmd = exports.handleOptions = void 0;
const tslib_1 = require("tslib");
const path_1 = require("path");
const util_args_1 = tslib_1.__importDefault(require("@git-lazy/util-args"));
const lodash_1 = require("lodash");
function handleOptions(remote, options) {
    var _a, _b;
    options = {
        ...options,
    };
    options.cwd = (_a = options.cwd) !== null && _a !== void 0 ? _a : process.cwd();
    if (((_b = options.targetDir) === null || _b === void 0 ? void 0 : _b.length) > 0) {
        options.targetDir = (0, path_1.resolve)(options.cwd, options.targetDir);
    }
    options = (0, lodash_1.defaultsDeep)(options, {
        cloneOptions: {
            depth: 50,
            singleBranch: true,
        },
    });
    return {
        remote,
        options,
    };
}
exports.handleOptions = handleOptions;
function gitCloneCmd(remote, options) {
    var _a, _b;
    ({ remote, options } = handleOptions(remote, options));
    const args = [
        'clone',
        remote,
    ];
    if (((_a = options.targetDir) === null || _a === void 0 ? void 0 : _a.length) > 0) {
        args.push(options.targetDir);
    }
    let ls = (0, util_args_1.default)((_b = options.cloneOptions) !== null && _b !== void 0 ? _b : {});
    if (ls.length) {
        args.push(...ls);
    }
    return args;
}
exports.gitCloneCmd = gitCloneCmd;
//# sourceMappingURL=util.js.map