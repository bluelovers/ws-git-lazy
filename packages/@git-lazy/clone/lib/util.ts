/**
 * Created by user on 2020/6/15.
 */

import { resolve } from "path";
import lazyUnParse from '@git-lazy/util-args';
import { IOptionsGitClone, IOptionsGitCloneSync } from './types';
import { defaultsDeep } from 'lodash';

export function handleOptions<T extends IOptionsGitClone | IOptionsGitCloneSync>(remote: string, options?: T)
{
	options = {
		...options,
	};

	options.cwd = options.cwd ?? process.cwd();

	if (options.targetDir?.length > 0)
	{
		options.targetDir = resolve(options.cwd, options.targetDir)
	}

	options = defaultsDeep(options, {
		cloneOptions: {
			depth: 50,
			singleBranch: true,
		},
	} as IOptionsGitClone)

	return {
		remote,
		options,
	}
}

export function gitCloneCmd(remote: string, options?: IOptionsGitClone)
{
	({ remote, options } = handleOptions(remote, options));

	const args = [
		'clone',
		remote,
	]

	if (options.targetDir?.length > 0)
	{
		args.push(options.targetDir)
	}

	let ls = lazyUnParse(options.cloneOptions ?? {})

	if (ls.length)
	{
		args.push(...ls)
	}

	return args
}
