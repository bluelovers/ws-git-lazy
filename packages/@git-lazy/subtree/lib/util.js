"use strict";
/**
 * Created by user on 2020/6/5.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.assertString = exports.handlePrefixPath = exports.inSubPath = exports.handlePrefix = void 0;
const upath2_1 = require("upath2");
const path_1 = require("path");
const types_1 = require("./types");
function handlePrefix(prefix) {
    const bool = /^\.[\/\\]/.test(prefix);
    prefix = (0, upath2_1.normalize)(prefix);
    if (bool && !prefix.startsWith('../')) {
        prefix = './' + prefix;
    }
    let prefixType = types_1.EnumPrefixType.ROOT;
    if (/^\/[^\/]/.test(prefix)) {
        prefix = prefix.slice(1);
    }
    else if (prefix.startsWith('../') || prefix.startsWith('./')) {
        prefixType = types_1.EnumPrefixType.RELATIVE;
    }
    else if ((0, path_1.isAbsolute)(prefix)) {
        prefixType = types_1.EnumPrefixType.ABSOLUTE;
    }
    return {
        prefixType,
        prefix,
    };
}
exports.handlePrefix = handlePrefix;
function inSubPath(sub, root) {
    let r = (0, upath2_1.normalize)(root);
    let s = (0, upath2_1.normalize)(sub);
    return s.indexOf(r) === 0 && s.length > r.length;
}
exports.inSubPath = inSubPath;
function handlePrefixPath(options) {
    let { prefix, prefixType, root, cwd, } = options;
    let prefixPath = prefix;
    if (prefixType !== types_1.EnumPrefixType.ROOT) {
        prefixPath = (0, upath2_1.resolve)(cwd, prefix);
        if (inSubPath(prefixPath, root)) {
            prefixPath = (0, upath2_1.relative)(root, prefixPath);
        }
        else {
            throw new Error(`prefix path is not allow: ${prefixPath}`);
        }
    }
    return {
        prefixPath,
        prefix,
        prefixType,
        root,
        cwd,
    };
}
exports.handlePrefixPath = handlePrefixPath;
function assertString(value, name) {
    if (typeof value !== 'string' || !value.length) {
        throw new TypeError(`${name !== null && name !== void 0 ? name : 'value'} is not valid: ${value}`);
    }
}
exports.assertString = assertString;
//# sourceMappingURL=util.js.map