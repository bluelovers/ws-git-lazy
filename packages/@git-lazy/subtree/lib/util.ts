/**
 * Created by user on 2020/6/5.
 */

import { normalize } from 'upath2';
import { isAbsolute } from "path";
import { EnumPrefixType } from './types';

export function handlePrefix(prefix: string)
{
	const bool = /^\.[\/\\]/.test(prefix);

	prefix = normalize(prefix);

	if (bool && !prefix.startsWith('../'))
	{
		prefix = './' + prefix;
	}

	let prefixType: EnumPrefixType = EnumPrefixType.ROOT;

	if (/^\/[^\/]/.test(prefix))
	{
		prefix = prefix.slice(1);
	}
	else if (prefix.startsWith('../') || prefix.startsWith('./'))
	{
		prefixType = EnumPrefixType.RELATIVE;
	}
	else if (isAbsolute(prefix))
	{
		prefixType = EnumPrefixType.ABSOLUTE;
	}

	return {
		prefixType,
		prefix,
	}
}

export function inSubPath(sub: string, root: string)
{
	let r = normalize(root)
	let s = normalize(sub)

	return s.indexOf(r) === 0 && s.length > r.length
}
