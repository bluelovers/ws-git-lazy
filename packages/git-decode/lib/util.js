"use strict";
/**
 * Created by user on 2020/6/4.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.wrapQuote = exports.removeQuote = exports.isQuote = void 0;
function isQuote(input) {
    return /^(["'])(.+)\1$/g.test(input);
}
exports.isQuote = isQuote;
function removeQuote(input) {
    return input.replace(/^(["'])(.+)\1$/g, '$2');
}
exports.removeQuote = removeQuote;
function wrapQuote(input, quote = '"') {
    if (!isQuote(input)) {
        return `${quote}${input}${quote}`;
    }
    return input;
}
exports.wrapQuote = wrapQuote;
//# sourceMappingURL=util.js.map