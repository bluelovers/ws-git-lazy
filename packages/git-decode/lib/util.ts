/**
 * Created by user on 2020/6/4.
 */

export function isQuote(input: string)
{
	return /^(["'])(.+)\1$/g.test(input)
}

export function removeQuote(input: string)
{
	return input.replace(/^(["'])(.+)\1$/g, '$2')
}

export function wrapQuote(input: string, quote = '"')
{
	if (!isQuote(input))
	{
		return `${quote}${input}${quote}`
	}

	return input
}
