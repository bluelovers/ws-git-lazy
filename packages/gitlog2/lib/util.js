"use strict";
/**
 * Created by user on 2019/1/6/006.
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.createError = exports.parseCommitsStdout = exports.parseCommits = exports.parseCommitFields = exports.addOptional = exports.addFlagsBool = exports.toFlag = exports.decamelize = exports.decode = exports.addPrettyFormat = exports.buildCommands = exports.handleOptions = exports.debug = void 0;
const tslib_1 = require("tslib");
const array_hyper_unique_1 = require("array-hyper-unique");
const debug_1 = tslib_1.__importDefault(require("debug"));
const fs_1 = require("fs");
const git_decode_1 = require("git-decode");
const type_1 = require("./type");
const lodash_assign_1 = tslib_1.__importDefault(require("lodash.assign"));
const decamelize_1 = tslib_1.__importDefault(require("decamelize"));
const sort_object_keys2_1 = require("sort-object-keys2");
const crlf_normalize_1 = require("crlf-normalize");
const const_1 = require("@git-lazy/const");
const stringify_1 = require("@lazy-spawn/stringify");
exports.debug = (0, debug_1.default)('gitlog');
function handleOptions(options) {
    var _a;
    var _b;
    // lazy name
    const REPO = (options.repo && options.repo != null) ? options.repo : options.cwd;
    if (!REPO)
        throw new Error(`Repo required!, but got "${REPO}"`);
    if (!(0, fs_1.existsSync)(REPO))
        throw new Error(`Repo location does not exist: "${REPO}"`);
    let defaultExecOptions = {
        cwd: REPO,
        stripAnsi: true,
    };
    // Set defaults
    options = (0, lodash_assign_1.default)({}, type_1.defaultOptions, { execOptions: defaultExecOptions }, options);
    options.execOptions = (0, lodash_assign_1.default)(options.execOptions, defaultExecOptions);
    (_a = (_b = options.execOptions).maxBuffer) !== null && _a !== void 0 ? _a : (_b.maxBuffer = const_1.GitExecMaxBuffer);
    if (options.returnAllFields) {
        options.fields = [].concat(Object.keys(type_1.fields));
        if (options.nameStatus && typeof options.nameStatusFiles == 'undefined') {
            options.nameStatusFiles = true;
        }
    }
    return options;
}
exports.handleOptions = handleOptions;
function buildCommands(options) {
    // Start constructing command
    let bin = 'git';
    let commands = [
        'log',
    ];
    commands = addFlagsBool(commands, options, [
        'findCopiesHarder',
        'all',
    ]);
    if (options.number > 0) {
        commands.push('-n', options.number);
    }
    commands = addFlagsBool(commands, options, [
        'noMerges',
        'firstParent',
    ]);
    if (options.displayFilesChangedDuringMerge) {
        commands.push('-m');
    }
    commands = addOptional(commands, options);
    commands = addPrettyFormat(commands, options, "pretty" /* EnumPrettyFormatFlags.PRETTY */);
    // Append branch (revision range) if specified
    if (options.branch) {
        commands.push(options.branch);
    }
    //File and file status
    commands = addFlagsBool(commands, options, [
        'nameStatus',
        'merges',
        'fullHistory',
        'sparse',
        'simplifyMerges',
    ]);
    if (options.file || options.files && options.files.length) {
        let ls = [options.file].concat(options.files || []).filter(v => v != null);
        if (!ls.length) {
            throw new TypeError(`file list is empty`);
        }
        commands = addFlagsBool(commands, options, [
            'follow',
        ]);
        commands.push('--', ...ls);
    }
    else if (options.follow) {
        throw new TypeError(`options.follow works only for a single file`);
    }
    (0, exports.debug)('command', options.execOptions, commands);
    return { bin, commands };
}
exports.buildCommands = buildCommands;
function addPrettyFormat(commands, options, flagName = "pretty" /* EnumPrettyFormatFlags.PRETTY */) {
    // Start of custom format
    // Iterating through the fields and adding them to the custom format
    let command = options.fields.reduce(function (command, field) {
        if (!type_1.fields[field] && type_1.notOptFields.indexOf(field) === -1)
            throw new RangeError('Unknown field: ' + field);
        command.push("\t" /* EnumPrettyFormatMark.DELIMITER */ + type_1.fields[field]);
        return command;
    }, [`${toFlag(flagName)}=${"@begin@" /* EnumPrettyFormatMark.BEGIN */}`])
        .concat(["@end@" /* EnumPrettyFormatMark.END */])
        .join("" /* EnumPrettyFormatMark.JOIN */);
    commands.push(command);
    return commands;
}
exports.addPrettyFormat = addPrettyFormat;
function decode(file) {
    if (file.indexOf('"') == 0 || file.match(/(?:\\(\d{3}))/)) {
        file = file.replace(/^"|"$/g, '');
        file = (0, git_decode_1.decode)(file);
    }
    return file;
}
exports.decode = decode;
function decamelize(key) {
    return (0, decamelize_1.default)(key, { separator: '-' });
}
exports.decamelize = decamelize;
function toFlag(key) {
    return '--' + decamelize(key);
}
exports.toFlag = toFlag;
function addFlagsBool(commands, options, flagNames) {
    for (let k of flagNames) {
        if (options[k]) {
            commands.push(toFlag(k));
        }
    }
    return commands;
}
exports.addFlagsBool = addFlagsBool;
/***
 Add optional parameter to command
 */
function addOptional(commands, options) {
    let cmdOptional = [
        'author',
        'since',
        'after',
        'until',
        'before',
        'committer',
        'skip',
    ];
    for (let k of cmdOptional) {
        if (options[k]) {
            commands.push(`--${k}=${options[k]}`);
        }
    }
    return commands;
}
exports.addOptional = addOptional;
function parseCommitFields(parsed, commitField, index, fields) {
    let key = fields[index];
    switch (key) {
        case 'tags':
            let tags = [];
            let start = commitField.indexOf('tag: ');
            if (start >= 0) {
                commitField
                    .substr(start + 5)
                    .trim()
                    .split(',')
                    .forEach(function (tag) {
                    tags.push(tag.trim());
                });
            }
            parsed[key] = tags;
            break;
        case 'authorDateUnixTimestamp':
        case 'committerDateUnixTimestamp':
            parsed[key] = parseInt(commitField);
            break;
        default:
            // @ts-ignore
            parsed[key] = commitField;
            break;
    }
    return parsed;
}
exports.parseCommitFields = parseCommitFields;
function parseCommits(commits, options) {
    let { fields, nameStatus } = options;
    return commits.map(function (_commit, _index) {
        //console.log(_commit);
        let parts = _commit.split("@end@" /* EnumPrettyFormatMark.END */);
        let commit = parts[0].split(type_1.delimiter);
        let nameStatusFiles = [];
        if (parts[1]) {
            let parseNameStatus = parts[1].trimLeft().split(crlf_normalize_1.LF);
            // Removes last empty char if exists
            if (parseNameStatus[parseNameStatus.length - 1] === '') {
                parseNameStatus.pop();
            }
            parseNameStatus = parseNameStatus
                // Split each line into it's own delimitered array
                .map(function (d, i) {
                return d.split(type_1.delimiter);
            })
                // 0 will always be status, last will be the filename as it is in the commit,
                // anything inbetween could be the old name if renamed or copied
                .reduce(function (a, b) {
                let tempArr = [b[0], b[b.length - 1]];
                tempArr[1] = decode(tempArr[1]);
                // @ts-ignore
                nameStatusFiles.push(tempArr);
                // If any files in between loop through them
                for (let i = 1, len = b.length - 1; i < len; i++) {
                    // If status R then add the old filename as a deleted file + status
                    // Other potentials are C for copied but this wouldn't require the original deleting
                    if (b[0].slice(0, 1) === type_1.EnumFileStatus.RENAMED) {
                        tempArr.push(type_1.EnumFileStatus.DELETED, b[i]);
                        // @ts-ignore
                        nameStatusFiles.push([type_1.EnumFileStatus.DELETED, decode(b[i])]);
                    }
                }
                return a.concat(tempArr);
            }, []);
            commit = commit.concat(parseNameStatus);
        }
        (0, exports.debug)('commit', commit);
        // Remove the first empty char from the array
        commit.shift();
        let parsed = {
            _index,
        };
        if (nameStatus) {
            // Create arrays for non optional fields if turned on
            type_1.notOptFields.forEach(function (d) {
                parsed[d] = [];
            });
        }
        commit.forEach(function (commitField, index) {
            if (fields[index]) {
                parsed = parseCommitFields(parsed, commitField, index, fields);
            }
            else {
                if (nameStatus) {
                    let pos = (index - fields.length) % type_1.notOptFields.length;
                    (0, exports.debug)('nameStatus', (index - fields.length), type_1.notOptFields.length, pos, commitField);
                    parsed[type_1.notOptFields[pos]].push(commitField);
                }
            }
        });
        if (nameStatus && options.nameStatusFiles) {
            parsed.fileStatus = (0, array_hyper_unique_1.array_unique)(nameStatusFiles);
        }
        // @ts-ignore
        parsed = (0, sort_object_keys2_1.sortObjectKeys)(parsed, type_1.KEY_ORDER);
        return parsed;
    });
}
exports.parseCommits = parseCommits;
function parseCommitsStdout(options, stdout) {
    let str;
    (0, exports.debug)('stdout', stdout);
    if (options.fnHandleBuffer) {
        str = options.fnHandleBuffer(stdout);
    }
    else {
        str = (0, stringify_1.crossSpawnOutput)(stdout);
    }
    //console.log(str);
    let commits = str.split("@begin@" /* EnumPrettyFormatMark.BEGIN */);
    if (commits[0] === '') {
        commits.shift();
    }
    (0, exports.debug)('commits', commits);
    commits = parseCommits(commits, options);
    (0, exports.debug)('commits:parsed', commits);
    return commits;
}
exports.parseCommitsStdout = parseCommitsStdout;
function createError(message, data, err) {
    // @ts-ignore
    err = err || Error;
    let e = message instanceof Error ? message : new err(message);
    // @ts-ignore
    e.data = data;
    // @ts-ignore
    return e;
}
exports.createError = createError;
//# sourceMappingURL=util.js.map