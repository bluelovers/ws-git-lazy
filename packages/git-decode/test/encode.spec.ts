import { decode, encode, decode2 } from '../index';
import { wrapQuote, removeQuote, isQuote } from '../lib/util';

const map = [
	[
		['\\351\\255\\224\\347\\245\\236\\350\\273\\242\\347\\224\\2372.CT'],
		'魔神転生2.CT',
	],
	[
		['66\\351\\255\\22477\\347\\245\\236\\350\\273\\242\\347\\224\\2372.CT'],
		'66魔77神転生2.CT',
	],
	[
		['"66\\351\\255\\22477\\347\\245\\236\\350\\273\\242\\347\\224\\2372.CT"'],
		'"66魔77神転生2.CT"',
	],
	[
		['\\302\\201'],
		String.fromCharCode(0x81),
	],
	[
		['\\351\\255\\224\\347\\245\\236\\350\\273\\242\\347\\224\\237'],
		'魔神転生',
	],
	[
		['\\302\\201\\351\\255\\224\\347\\245\\236\\350\\273\\242\\347\\224\\237'],
		String.fromCharCode(0x81) + '魔神転生',
	],
	[
		['\\351\\255\\224\\347\\245\\236\\350\\273\\242\\347\\224\\237\\302\\201'],
		'魔神転生' + String.fromCharCode(0x81),
	],
	[
		['999 \\346\\226\\260\\346\\226\\207\\345\\255\\227\\346\\226\\207\\177\\344\\273\\266.TXT'],
		'999 新文字文件.TXT',
	],
	[
		['999 { }  \' ~.TXT'],
		'999 { }  \' ~.TXT',
	],
	[
		['\\177'],
		'',
	],
	[
		['2\\177'],
		'2',
	],
	[
		['\\177.'],
		'.',
	],
	[
		['\\342\\230\\272'],
		'☺',
	],
	[
		['999 \\346\\265\\205\\350\\260\\210 JS \\357\\274\\241\\357\\274\\242 \\345\\257\\271\\350\\261\\241\\344\\271\\213\\346\\211\\251\\345\\261\\225.TXT'],
		'999 浅谈 JS ＡＢ 对象之扩展.TXT',
	],
] as [[string], string][];

describe(`decode`, () =>
{

	map.forEach(function (v)
		{
			test(`${v[1]} = ${v[0][0]}`, () =>
				{
					let actual = decode(...v[0]);
					expect(actual).toStrictEqual(v[1]);
					expect(actual).toMatchSnapshot();
				},
			);
		},
	)

})

describe('encode', function ()
	{
		map.forEach(function (v)
			{
				it(`${v[1]} = ${v[0][0]}`, () =>
					{
						let actual = encode(v[1]);
						expect(actual).toStrictEqual(v[0][0]);
						expect(actual).toMatchSnapshot();
					},
				);
			},
		)
	},
);

describe('encode = decode', function ()
	{
		map.forEach(function (v)
			{
				it(`${v[1]} = ${v[0][0]}`, () =>
					{
						let actual = decode(v[0][0]);

						expect(actual).toStrictEqual(decode(encode(actual)));
						expect(actual).toStrictEqual(v[1]);

						actual = encode(v[1]);

						expect(actual).toStrictEqual(encode(decode(actual)));
						expect(actual).toStrictEqual(v[0][0]);
					},
				);
			},
		)
	},
);

describe(`decode2`, () =>
{

	map.forEach(function (v)
		{
			let s = wrapQuote(`${v[0][0]}`, '\'')
			let expected = removeQuote(v[1]);

			test(`${expected} = ${s}`, () =>
				{
					let actual = decode2(s);
					expect(actual).toStrictEqual(expected);
					expect(isQuote(actual)).toStrictEqual(false);
					expect(actual).toMatchSnapshot();
				},
			);

			let s2 = wrapQuote(`${v[0][0]}`)

			test(`${expected} = ${s2}`, () =>
				{
					let actual = decode2(s2);
					expect(actual).toStrictEqual(expected);
					expect(isQuote(actual)).toStrictEqual(false);
					expect(actual).toMatchSnapshot();
				},
			);
		},
	)

})
