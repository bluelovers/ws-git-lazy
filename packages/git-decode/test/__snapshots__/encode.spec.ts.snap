// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`decode "66魔77神転生2.CT" = "66\\351\\255\\22477\\347\\245\\236\\350\\273\\242\\347\\224\\2372.CT" 1`] = `""66魔77神転生2.CT""`;

exports[`decode  = \\177 1`] = `""`;

exports[`decode . = \\177. 1`] = `"."`;

exports[`decode 2 = 2\\177 1`] = `"2"`;

exports[`decode 66魔77神転生2.CT = 66\\351\\255\\22477\\347\\245\\236\\350\\273\\242\\347\\224\\2372.CT 1`] = `"66魔77神転生2.CT"`;

exports[`decode 999 { }  ' ~.TXT = 999 { }  ' ~.TXT 1`] = `"999 { }  ' ~.TXT"`;

exports[`decode 999 新文字文件.TXT = 999 \\346\\226\\260\\346\\226\\207\\345\\255\\227\\346\\226\\207\\177\\344\\273\\266.TXT 1`] = `"999 新文字文件.TXT"`;

exports[`decode 999 浅谈 JS ＡＢ 对象之扩展.TXT = 999 \\346\\265\\205\\350\\260\\210 JS \\357\\274\\241\\357\\274\\242 \\345\\257\\271\\350\\261\\241\\344\\271\\213\\346\\211\\251\\345\\261\\225.TXT 1`] = `"999 浅谈 JS ＡＢ 对象之扩展.TXT"`;

exports[`decode  = \\302\\201 1`] = `""`;

exports[`decode 魔神転生 = \\302\\201\\351\\255\\224\\347\\245\\236\\350\\273\\242\\347\\224\\237 1`] = `"魔神転生"`;

exports[`decode ☺ = \\342\\230\\272 1`] = `"☺"`;

exports[`decode 魔神転生 = \\351\\255\\224\\347\\245\\236\\350\\273\\242\\347\\224\\237 1`] = `"魔神転生"`;

exports[`decode 魔神転生2.CT = \\351\\255\\224\\347\\245\\236\\350\\273\\242\\347\\224\\2372.CT 1`] = `"魔神転生2.CT"`;

exports[`decode 魔神転生 = \\351\\255\\224\\347\\245\\236\\350\\273\\242\\347\\224\\237\\302\\201 1`] = `"魔神転生"`;

exports[`decode2  = "\\177" 1`] = `""`;

exports[`decode2  = '\\177' 1`] = `""`;

exports[`decode2 . = "\\177." 1`] = `"."`;

exports[`decode2 . = '\\177.' 1`] = `"."`;

exports[`decode2 2 = "2\\177" 1`] = `"2"`;

exports[`decode2 2 = '2\\177' 1`] = `"2"`;

exports[`decode2 66魔77神転生2.CT = "66\\351\\255\\22477\\347\\245\\236\\350\\273\\242\\347\\224\\2372.CT" 1`] = `"66魔77神転生2.CT"`;

exports[`decode2 66魔77神転生2.CT = "66\\351\\255\\22477\\347\\245\\236\\350\\273\\242\\347\\224\\2372.CT" 2`] = `"66魔77神転生2.CT"`;

exports[`decode2 66魔77神転生2.CT = "66\\351\\255\\22477\\347\\245\\236\\350\\273\\242\\347\\224\\2372.CT" 3`] = `"66魔77神転生2.CT"`;

exports[`decode2 66魔77神転生2.CT = '66\\351\\255\\22477\\347\\245\\236\\350\\273\\242\\347\\224\\2372.CT' 1`] = `"66魔77神転生2.CT"`;

exports[`decode2 999 { }  ' ~.TXT = "999 { }  ' ~.TXT" 1`] = `"999 { }  ' ~.TXT"`;

exports[`decode2 999 { }  ' ~.TXT = '999 { }  ' ~.TXT' 1`] = `"999 { }  ' ~.TXT"`;

exports[`decode2 999 新文字文件.TXT = "999 \\346\\226\\260\\346\\226\\207\\345\\255\\227\\346\\226\\207\\177\\344\\273\\266.TXT" 1`] = `"999 新文字文件.TXT"`;

exports[`decode2 999 新文字文件.TXT = '999 \\346\\226\\260\\346\\226\\207\\345\\255\\227\\346\\226\\207\\177\\344\\273\\266.TXT' 1`] = `"999 新文字文件.TXT"`;

exports[`decode2 999 浅谈 JS ＡＢ 对象之扩展.TXT = "999 \\346\\265\\205\\350\\260\\210 JS \\357\\274\\241\\357\\274\\242 \\345\\257\\271\\350\\261\\241\\344\\271\\213\\346\\211\\251\\345\\261\\225.TXT" 1`] = `"999 浅谈 JS ＡＢ 对象之扩展.TXT"`;

exports[`decode2 999 浅谈 JS ＡＢ 对象之扩展.TXT = '999 \\346\\265\\205\\350\\260\\210 JS \\357\\274\\241\\357\\274\\242 \\345\\257\\271\\350\\261\\241\\344\\271\\213\\346\\211\\251\\345\\261\\225.TXT' 1`] = `"999 浅谈 JS ＡＢ 对象之扩展.TXT"`;

exports[`decode2  = "\\302\\201" 1`] = `""`;

exports[`decode2  = '\\302\\201' 1`] = `""`;

exports[`decode2 魔神転生 = "\\302\\201\\351\\255\\224\\347\\245\\236\\350\\273\\242\\347\\224\\237" 1`] = `"魔神転生"`;

exports[`decode2 魔神転生 = '\\302\\201\\351\\255\\224\\347\\245\\236\\350\\273\\242\\347\\224\\237' 1`] = `"魔神転生"`;

exports[`decode2 ☺ = "\\342\\230\\272" 1`] = `"☺"`;

exports[`decode2 ☺ = '\\342\\230\\272' 1`] = `"☺"`;

exports[`decode2 魔神転生 = "\\351\\255\\224\\347\\245\\236\\350\\273\\242\\347\\224\\237" 1`] = `"魔神転生"`;

exports[`decode2 魔神転生 = '\\351\\255\\224\\347\\245\\236\\350\\273\\242\\347\\224\\237' 1`] = `"魔神転生"`;

exports[`decode2 魔神転生2.CT = "\\351\\255\\224\\347\\245\\236\\350\\273\\242\\347\\224\\2372.CT" 1`] = `"魔神転生2.CT"`;

exports[`decode2 魔神転生2.CT = '\\351\\255\\224\\347\\245\\236\\350\\273\\242\\347\\224\\2372.CT' 1`] = `"魔神転生2.CT"`;

exports[`decode2 魔神転生 = "\\351\\255\\224\\347\\245\\236\\350\\273\\242\\347\\224\\237\\302\\201" 1`] = `"魔神転生"`;

exports[`decode2 魔神転生 = '\\351\\255\\224\\347\\245\\236\\350\\273\\242\\347\\224\\237\\302\\201' 1`] = `"魔神転生"`;

exports[`encode "66魔77神転生2.CT" = "66\\351\\255\\22477\\347\\245\\236\\350\\273\\242\\347\\224\\2372.CT" 1`] = `""66\\351\\255\\22477\\347\\245\\236\\350\\273\\242\\347\\224\\2372.CT""`;

exports[`encode  = \\177 1`] = `"\\177"`;

exports[`encode . = \\177. 1`] = `"\\177."`;

exports[`encode 2 = 2\\177 1`] = `"2\\177"`;

exports[`encode 66魔77神転生2.CT = 66\\351\\255\\22477\\347\\245\\236\\350\\273\\242\\347\\224\\2372.CT 1`] = `"66\\351\\255\\22477\\347\\245\\236\\350\\273\\242\\347\\224\\2372.CT"`;

exports[`encode 999 { }  ' ~.TXT = 999 { }  ' ~.TXT 1`] = `"999 { }  ' ~.TXT"`;

exports[`encode 999 新文字文件.TXT = 999 \\346\\226\\260\\346\\226\\207\\345\\255\\227\\346\\226\\207\\177\\344\\273\\266.TXT 1`] = `"999 \\346\\226\\260\\346\\226\\207\\345\\255\\227\\346\\226\\207\\177\\344\\273\\266.TXT"`;

exports[`encode 999 浅谈 JS ＡＢ 对象之扩展.TXT = 999 \\346\\265\\205\\350\\260\\210 JS \\357\\274\\241\\357\\274\\242 \\345\\257\\271\\350\\261\\241\\344\\271\\213\\346\\211\\251\\345\\261\\225.TXT 1`] = `"999 \\346\\265\\205\\350\\260\\210 JS \\357\\274\\241\\357\\274\\242 \\345\\257\\271\\350\\261\\241\\344\\271\\213\\346\\211\\251\\345\\261\\225.TXT"`;

exports[`encode  = \\302\\201 1`] = `"\\302\\201"`;

exports[`encode 魔神転生 = \\302\\201\\351\\255\\224\\347\\245\\236\\350\\273\\242\\347\\224\\237 1`] = `"\\302\\201\\351\\255\\224\\347\\245\\236\\350\\273\\242\\347\\224\\237"`;

exports[`encode ☺ = \\342\\230\\272 1`] = `"\\342\\230\\272"`;

exports[`encode 魔神転生 = \\351\\255\\224\\347\\245\\236\\350\\273\\242\\347\\224\\237 1`] = `"\\351\\255\\224\\347\\245\\236\\350\\273\\242\\347\\224\\237"`;

exports[`encode 魔神転生2.CT = \\351\\255\\224\\347\\245\\236\\350\\273\\242\\347\\224\\2372.CT 1`] = `"\\351\\255\\224\\347\\245\\236\\350\\273\\242\\347\\224\\2372.CT"`;

exports[`encode 魔神転生 = \\351\\255\\224\\347\\245\\236\\350\\273\\242\\347\\224\\237\\302\\201 1`] = `"\\351\\255\\224\\347\\245\\236\\350\\273\\242\\347\\224\\237\\302\\201"`;
